File: npm/examples/basic.js
/**
 * Basic Example - UTM Coordinate Converter API
 *
 * This example demonstrates how to use the UTM Coordinate Converter API.
 * Make sure to set your API key in the .env file or replace '[YOUR_API_KEY]' below.
 */

require('dotenv').config();
const utmconverterAPI = require('../index.js');

// Initialize the API client
const api = new utmconverterAPI({
    api_key: process.env.API_KEY || '[YOUR_API_KEY]'
});

// Example query
var query = {
  direction: "to_utm",
  latitude: 40.7128,
  longitude: -74.0060,
  zone: 18,
  easting: 585628,
  northing: 4511322,
  hemisphere: "N"
};

// Make the API request using callback
console.log('Making request to UTM Coordinate Converter API...\n');

api.execute(query, function (error, data) {
    if (error) {
        console.error('Error occurred:');
        if (error.error) {
            console.error('Message:', error.error);
            console.error('Status:', error.status);
        } else {
            console.error(JSON.stringify(error, null, 2));
        }
        return;
    }

    console.log('Response:');
    console.log(JSON.stringify(data, null, 2));
});

File: examples/python/basic.py
"""
UTM Coordinate Converter API - Basic Usage Example

This example demonstrates the basic usage of the UTM Coordinate Converter API.
API Documentation: https://docs.apiverve.com/ref/utmconverter
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/utmconverter'

def call_utmconverter_api():
    """
    Make a GET request to the UTM Coordinate Converter API
    """
    try:
        # Query parameters
        params &#x3D; {&#x27;direction&#x27;: &#x27;to_utm&#x27;, &#x27;latitude&#x27;: 40.7128, &#x27;longitude&#x27;: -74.006}

        headers = {
            'x-api-key': API_KEY
        }

        response = requests.get(API_URL, headers=headers, params=params)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling UTM Coordinate Converter API...\n')

    result = call_utmconverter_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')
